modules/nextflow/src/main/groovy/nextflow/scm/AssetManager.groovy

```diff
@@ -542,16 +565,39 @@ class AssetManager {
         return result
     }
 
-    static protected def find( String name ) {
+    /**
+     * @return The list of available revisions for a given project name
+     */
+    List<String> listRevisions( String projectName = project ) {
+        log.debug "Listing revisions for project: $projectName"
+
+        def result = new LinkedList()
+        if( !root.exists() )
+            return result
+
+        list().each {
+            if( it.tokenize(revisionDelim)[0] == projectName ) {
+                result << it
+            }
+        }
+
+        return result
+    }
+
+    // updated for new localPath schema (see localPath declaration at top of this class file)
+    static protected def find( String name, String revision = null ) {
         def exact = []
         def partial = []
 
         list().each {
             def items = it.split('/')
-            if( items[1] == name )
-                exact << it
-            else if( items[1].startsWith(name ) )
-                partial << it
+            def itemsRev = items[1].tokenize(':')
+            if( (!revision && !itemsRev[1]) || (revision && itemsRev[1] == revision) ) {
+                if( itemsRev[0] == name )
+                    exact << it.tokenize(':')[0]
```

Comment:
This `:` should not be `revisionDelim` ?

Reply:
thanks, fixed